plugins/modules/keycloak_realm_localization.py

```diff
@@ -0,0 +1,391 @@
+from copy import deepcopy
+
+
+def _normalize_overrides_from_api(current):
+    """
+    Accept either:
+      - dict: {'k1': 'v1', ...}
+      - list of dicts: [{'key': 'k1', 'value': 'v1'}, ...]
+    Return a sorted list of {'key', 'value'}.
+
+    This helper provides a consistent shape for downstream comparison/diff logic.
+    """
+    if not current:
+        return []
+
+    if isinstance(current, dict):
+        # Convert mapping to list of key/value dicts
+        items = [{'key': k, 'value': v} for k, v in current.items()]
+    else:
+        # Assume a list of dicts with keys 'key' and 'value'
+        items = [{'key': o['key'], 'value': o.get('value')} for o in current]
+
+    # Sort for stable comparisons and diff output
+    return sorted(items, key=lambda x: x['key'])
+
+
+def main():
+    """
+    Module execution
+
+    :return:
+    """
+    # Base Keycloak auth/spec fragment common across Keycloak modules
+    argument_spec = keycloak_argument_spec()
+
+    # Describe a single override record
+    overrides_spec = dict(
+        key=dict(type='str', no_log=False, required=True),
+        value=dict(type='str', required=True),
+    )
+
+    # Module-specific arguments
+    meta_args = dict(
+        locale=dict(type='str', required=True),
+        parent_id=dict(type='str', required=True),
+        state=dict(type='str', default='present', choices=['present', 'absent']),
+        overrides=dict(type='list', elements='dict', options=overrides_spec, default=[]),
+    )
+
+    argument_spec.update(meta_args)
+
+    module = AnsibleModule(argument_spec=argument_spec,
+                           supports_check_mode=True,
+                           # Require token OR full credential set. This mirrors other Keycloak modules.
+                           required_one_of=([['token', 'auth_realm', 'auth_username', 'auth_password']]),
+                           required_together=([['auth_realm', 'auth_username', 'auth_password']]))
+
+    # Initialize the result object used by Ansible
+    result = dict(changed=False, msg='', end_state={}, diff=dict(before={}, after={}))
+
+    # Obtain access token, initialize API
+    try:
+        connection_header = get_token(module.params)
+    except KeycloakError as e:
+        module.fail_json(msg=str(e))
+
+    kc = KeycloakAPI(module, connection_header)
+
+    # Convenience locals for frequently used parameters
+    locale = module.params.get('locale')
+    state = module.params.get('state')
+    parent_id = module.params.get('parent_id')
+
+    # Desired overrides: deduplicate by key via dict (last wins), then sort
+    desired_raw = module.params.get('overrides') or []
+    if state == 'present':
+        # Validate that all keys have a value in present mode
+        missing_values = [r['key'] for r in desired_raw if r.get('value') is None]
+        if missing_values:
+            module.fail_json(msg="state=present requires 'value' for keys: %s" % ", ".join(missing_values))
```

Comment:
Given that param `overrides` makes both `key`and `value` required (lines 233-234), this entire check is redundant.

#sigh #aifeelings